src/commands/createWebApp/createWebApp.ts

```diff
@@ -3,24 +3,47 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AzureParentTreeItem, IActionContext, ICreateChildImplContext } from "vscode-azureextensionui";
+import { MessageItem, window } from "vscode";
+import { AzureParentTreeItem, callWithTelemetryAndErrorHandling, IActionContext, ICreateChildImplContext, parseError } from "vscode-azureextensionui";
 import { SubscriptionTreeItem } from '../../explorer/SubscriptionTreeItem';
 import { WebAppTreeItem } from "../../explorer/WebAppTreeItem";
 import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
 import { showCreatedWebAppMessage } from "./showCreatedWebAppMessage";
 
-export async function createWebApp(context: IActionContext & Partial<ICreateChildImplContext>, node?: AzureParentTreeItem | undefined, suppressCreatedWebAppMessage: boolean = false): Promise<WebAppTreeItem> {
+export async function createWebApp(context: IActionContext & Partial<ICreateChildImplContext>, node?: AzureParentTreeItem | undefined, suppressCreatedWebAppMessage: boolean = false): Promise<WebAppTreeItem | undefined> {
     if (!node) {
         node = <AzureParentTreeItem>await ext.tree.showTreeItemPicker(SubscriptionTreeItem.contextValue, context);
     }
-
-    const newSite: WebAppTreeItem = <WebAppTreeItem>await node.createChild(context);
-    if (!suppressCreatedWebAppMessage) {
-        showCreatedWebAppMessage(newSite);
+    let newSite: WebAppTreeItem;
+    const authErrorRegexp = new RegExp(`The client '.+' with object id '.+' does not have authorization to perform action 'Microsoft\.Web\/serverfarms\/read' over scope '.+' or the scope is invalid\. If access was recently granted, please refresh your credentials\.`);
+    try {
+        newSite = <WebAppTreeItem>await node.createChild(context);
+        if (!suppressCreatedWebAppMessage) {
+            showCreatedWebAppMessage(newSite);
+        }
+        return newSite;
+    } catch (error) {
+        if (!parseError(error).message.match(authErrorRegexp)) {
+            throw error;
+        } else {
+            const message = localize('notAuthorizedToCreateRGs', 'You subscription is not authorized to create resource groups. Please create using Advanced Create to select existing resource groups.')
+            const advancedCreate: MessageItem = { title: localize('createAdvanced', 'Create - Advanced') };
+            // don't wait
+            void window.showInformationMessage(message, advancedCreate).then(async (result: MessageItem | undefined) => {
```

Comment:
We should be able to come up with a solution that keeps the user in basic create. For example, we're generating the plan name ourselves, so maybe we just generate a new name that doesn't conflict with the existing plan they don't have access to.

I think your first priority should be to figure out how to recreate the exact error yourself. You might need someone else to give you partial access to some resources on their subscription. That should help you pinpoint where the actual error is occurring and how to fix it within the basic create wizard.

Reply:
I have been testing this scenario using Sandbox accounts and narrowed down the issue to be happening at [`tryGetAppServicePlan(...)`](https://github.com/microsoft/vscode-azureappservice/blob/8b750a07d0f741982f475ac02ca3f98fff4954b4/src/commands/createWebApp/setPostPromptDefaults.ts#L42).

I haven't tested scenarios other than Sandbox. I will check if those account scenarios lead to the same breakpoint.